### PR TITLE
Feature/test external download

### PR DESCRIPTION
**Pull request by tonsV2**

### CODE REVIEW

File: pkg/database/handler_test.go [hunk 1]
```diff
@@ -26,6 +26,53 @@ import (
...
+		Return(database, nil)
+	repository.
+		On("PurgeExternalDownload").
+		Return(nil)
+	service := NewService(config.Config{}, nil, s3Client, repository)
+	handler := New(nil, service, nil)
+
+	w := httptest.NewRecorder()
+	c := newContext(w, "group-name")
+	c.AddParam("uuid", uuid.New().String())
```
Comment (teleivo) [inline pkg/database/handler_test.go:61]: How does this uuid relate to what DB we are downloading? To me a test should clearly show the connection between the inputs to the method we test and either an output or some behavior. I can't see it 😅

Comment (tonsV2) [reply pkg/database/handler_test.go:61]: https://github.com/dhis2-sre/im-database-manager/blob/master/pkg/database/handler.go#L755

It's the PK of the ExternalDownload model... https://github.com/dhis2-sre/im-database-manager/blob/cc35cfa63f63081722fdd90a64717683bd74a6eb/pkg/model/database.go#L26

I'm not sure how to convey that :relaxed:

Comment (teleivo) [reply pkg/database/handler_test.go:61]: right, it has to match the uuid in your mock


```go
 		On("FindExternalDownload", mock.AnythingOfType("uuid.UUID")).
```

so for me the test should make that clear and assert that is the case

Comment (tonsV2) [reply pkg/database/handler_test.go]: Oh yeah, I try to avoid mock.AnythingOfType as much as possible and now it's no longer used here

**Review by teleivo — APPROVED** [review]

**Pull request merged**